astropy/modeling/core.py

```diff
@@ -805,6 +805,16 @@ def n_outputs(self):
 
         return self.__class__.n_outputs
 
+    @classmethod
+    def _calculate_separability_matrix(cls):
+        """
+        This is a hook which customises the behavior of modeling.separable.
+
+        This allows complex subclasses to customise the separability matrix.
+        If it returns `NotImplemented` the default behavior is used.
+        """
+        return NotImplemented
```

Comment:
In your linked example this is actually not a `classmethod`. Should it be a property or just a regular method instead?

Reply:
Yeah, I am unclear if `separability_matrix` only accepts instances?